### DISCUSSION TITLE
windows flags explorerpatcher as malware in latest update (marked as high severity)

### DISCUSSION THREAD

**Discussion by thesussychanelgithub** [valinet/ExplorerPatcher#3122] [General]
in the latest version of windows, explorer patcher is flagged as malware by defender and instantly deleted. i had explorer patcher for a while now and it never started appearing as a virus until now. see the pictures i have here
![Screenshot 2024-04-10 202434](https://github.com/valinet/ExplorerPatcher/assets/166666176/73475d52-94a3-43eb-aa8d-cd07fc805300)
![Screenshot 2024-04-10 202516](https://github.com/valinet/ExplorerPatcher/assets/166666176/b6ba31e8-8df1-480d-913b-88d918a8b907)

**Comment by thesussychanelgithub** [comment]
it tried to delete it AGAIN while i was writing this

**Comment by iwan1979** [comment]
Indeed the latest pre-release version 22621.3296.64.3 was detected by 11 security tools as malware while the previous three versions (22621.3296.64.2 and 22621.3296.64.1) have low false positives as you can refer to below links and screenshots. it makes me hesitate to try out the latest pre-release version.

22621.3296.64.3
https://www.virustotal.com/gui/file/672d25a8c753423ddf0622a7191dfa803093528e44d05fa03b94cefc42196199?nocache=1
![20240411_124311](https://github.com/valinet/ExplorerPatcher/assets/20137476/c4f4fb83-58d1-445f-9322-05978c2b0f8c)


22621.3296.64.2
https://www.virustotal.com/gui/file/d5f33b9a84c0ffe7b7361012f267316893c7559ad90b68c1f4af75bc0fbfb324?nocache=1
![20240411_124306](https://github.com/valinet/ExplorerPatcher/assets/20137476/908b984e-4b3a-483b-bd4e-fa2e8fe1db49)


22621.3296.64.2
https://www.virustotal.com/gui/file/2783a9b68066521b52a82d5a47d8c683cb363cdfcdc6cb578ad4b223595fd6fd?nocache=1
![20240411_124301](https://github.com/valinet/ExplorerPatcher/assets/20137476/56fcc557-a95f-4081-9dad-e1e66706048f)


22621.3296.64.1
https://www.virustotal.com/gui/file/167f90fad0d3df4e2bfd4a6b07e459a6e414f63fa20e473aedc8c82f3b21eaef?nocache=1
![20240411_124248](https://github.com/valinet/ExplorerPatcher/assets/20137476/58c03b02-c74e-4c8c-aa2a-0f948a20c700)

**Comment by pyrates999** [comment]
This is a false positive.

Update your windows defender signatures by going to settings -> updates & security -> windows security
Then click on virus & threat protection
Under virus & threat protection updates, click on check for updates
Then click on the button named check for updates

Then if it is still being detected as malicious, submit it as a false positive to Microsoft: https://www.microsoft.com/en-us/wdsi/filesubmission

**Comment by TheAncient2** [comment]
The more people submit this as a false positive to Microsoft, the better the chances they will fix it.
I suggest you also let them know that this happens frequently when a new update for ExplorerPatcher is released.
Maybe, just maybe, one of the geniuses at Microfluff will see a trend here and devise a permanent fix.

**Comment by V7614** [comment]
Hi I am not a developer of explorerpatcher but i am also a explorerpatcher user like you
So t think it is because Microsoft does not want you to use the new taskbar implementation.
 there is an easy workaround though here it is: 
1. download a browser example: firefox or google chrome (i don't think edge will allow you to)
2. install an antivirus (I am using Malwarebytes) 
3. install explorerpatcher setup once you open it Malwarebytes will quarantine it just go to detection history on the dashboard as shown in the picture below:
![Malwarebytes detection history](https://github.com/user-attachments/assets/5a6874ae-f9c1-4b69-aaa5-d6a20b0067d4)
2a. select the setup and click restore shown in picture (do the one in the red square first before the one in green square) and if it asks click yes:
![Restore setup](https://github.com/user-attachments/assets/1830c74f-bc39-4aa0-9147-5753ed4184df)
3. click allow list and add the setup file which is called ep_setup (I show the file extensions) like below:
![Add item to Malwarebytes allow list](https://github.com/user-attachments/assets/6be888ad-2328-4208-914f-085b53c91b96)
![Add item to Malwarebytes allow list(1)](https://github.com/user-attachments/assets/357c7714-811b-45c0-9663-321d2c86ac99)
![Add item to Malwarebytes allow list(2)](https://github.com/user-attachments/assets/e44cc7b9-b6f2-4fb2-a729-c067c12d11e7)
![Add item to Malwarebytes allow list(3)](https://github.com/user-attachments/assets/3d4a15cd-a7ac-457a-ad67-5d38b890f8f2)
![Add item to Malwarebytes allow list(4)](https://github.com/user-attachments/assets/0e1f1882-6ca0-4315-9a46-eea474c96f57)
it should look something like this:
![Add item to Malwarebytes allow list(5)](https://github.com/user-attachments/assets/0eb7a660-2153-445b-a369-b555e9831572)
4. Open it again and it should work

(I just saw this post today through google search hope this helps)

**Reply by crack-kitty** [reply]
Just FYI, the same thing works for windows defender.

Windows Security
Virus & Threat Protection
Virus & Threat Protection Settings (Manage Settings)
Scroll to bottom - Exclusions (Add or remove exclusions)

I just added my downloads folder, completed the download.
Windows yelled about the file.
More info.
Run anyway.

Worked for me on W11

**Reply by OldFox-62** [reply]
this works for me for Microsoft Defender
add folders to allow list:

C:\Program Files\ExporerPatcher
C:\Users\\{USERNAME}\AppData\Roaming\ExporerPatcher

**Comment by Just2Confuse** [comment]
My problem trusting this is that Virustotal is completely going crazy. This is not only a "Microsoftproblem". 
If there are one or 3 noname virusprogramms? No problem, but 32 from 76?
https://www.virustotal.com/gui/file/1c4e1847c722db18d58216c43aa40ad87c8a38aa6196e69d55c0687b8506bf94?nocache=1

Jotti Malwarescan get 6 of 16. 
I'll wait until things have calmed down. Not that I don't trust the whole thing, but it's never been this bad before.

**Reply by iwan1979** [reply]
Hi, 

I have the same concern as urs. it's not just microsoft defender issue only. till now i still stick with v64.2 given that's the latest version with lesser false positives (only 2-3)

**Comment by V7614** [comment]
that did not work for me so i switched

**Comment by Menno5** [comment]
Kaspersky here also, flagging EP as _UDS:DangerousObject.Multi.Generic_.
Having said that: I'm almost sure that all anti virus programs are reacting on the fact that Microsoft flagged EP as "dangerous". Most likely in preparation for releasing 24H2 in the wild. Microsoft doesn't like programs like EP.
Added EP in the Ignore List both in MS Defender and Kaspersky.

**Comment by pyrates999** [comment]
Discussed here already: https://github.com/valinet/ExplorerPatcher/issues/3228

MS just doesn't like EP and has added it to windows defender, even though it is not a virus and not a hack tool.

See the release notes here for the powershell script you can run that excludes all the directories necessary in windows defender: https://github.com/valinet/ExplorerPatcher/releases

The powershell script you must run with elevated privileges.

Future updates to EP won't be flagged then.

You can also set windows defender to exclude the directory that you manually download EP to so you can install it without windows defender blocking it.

**Reply by descentfan1975** [reply]
That worked and I was able to install the update so I'm now running Version 22621.3880.66.5, but now when I tell it to check for updates, it gives me an error message that it is unable to check for updates and that I should check to see if I'm connected to the internet (I am).

I'm on Windows 23H2, build 22631.4169

**Reply by kslifter** [reply]
Thanks for the link to the Release Notes and the Powershell script. It worked for me.

**Comment by Laura-Ess** [comment]
After attempting some of these I had to uninstall _ExplorerPatcher_. It was causing issues with windows and those issues were affecting another app called _DreamGrid_ (an enhanced version of _Open Simulator_). I lost a fifth of the data in that app! I had the same issues on my other Win PCs and rather than struggle this, which was driving me to panic attacks and depression, I've uninstalled ExplorePatcher for now. 

Am NOT HAPPY about that but it's the only way for me not to be driven nuts by this. I really liked ExplorerPatcher and its features, but now I'm going back to a desktop that seems to be a half arsed imitation of a Mac. NOT HAPPY, but thank you for your efforts on this - they may my using Win 11 a LOT EASIER.

**Reply by iwan1979** [reply]
probably you want to use back the previous version that stable for your workstation. for me, it's v64.2

**Reply by pyrates999** [reply]
you may have enabled the option register as shell extension, disable it by opening up the EP properties application, go to **File Explorer**, and untick **Register as shell extension**.

Here's why: https://github.com/valinet/ExplorerPatcher/wiki/Using-ExplorerPatcher-as-shell-extension

**Reply by Laura-Ess** [reply]
I'll try those options after I recover the OpenSim Grid. Thanks for the suggestions.

**Comment by arcyc** [comment]
In the latest update installed today there is an **actual** virus detected.  This is exactly the pattern used to trick people into installing malware - train them to ignore the error, then slip in the malware.
<img width="373" alt="image" src="https://github.com/user-attachments/assets/f99fde9c-29ef-4378-b015-3bd8826aa729">

**Reply by Just2Confuse** [reply]
The latest Release looks better, some more as any other before. 
https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3880.66.6_92fce8c

https://www.virustotal.com/gui/file-analysis/NDVhNWE0NDNjMDFhYmQ3NjE4ZWZlZjQ4MjcyNDEzMTI6MTcyNjIzMjIwMQ==

**Reply by arcyc** [reply]
Only 9 malwares 🙂.  The repo has obviously been compromised at this point.  Time to wipe some machines.

**Reply by Just2Confuse** [reply]
> Only 9 malwares 🙂. The repo has obviously been compromised at this point. Time to wipe some machines.

Thats bullshit. There some False-Positive in every relase since years. Calm down and  stop spreading such nonsense if you don't know what you're seeing.

**Reply by iwan1979** [reply]
the creators of this great tool never force anyone to use it. we just need to exercise cautious. hope the false positives gonna further reduced

**Reply by iwan1979** [reply]
@Just2Confuse 

Have you installed this latest v66.6? look promising with 10 false positives and nothing detected by microsoft antivirus! 
![VirusTotal-File-d7f98b8af8a3bfe9d93ce31558a62e4d5d0cd425bc30bbc0d517901e5b82bf46](https://github.com/user-attachments/assets/672cf33d-35a9-41dd-91ae-e31286e67037)


> The latest Release looks better, some more as any other before. https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3880.66.6_92fce8c
> 
> https://www.virustotal.com/gui/file-analysis/NDVhNWE0NDNjMDFhYmQ3NjE4ZWZlZjQ4MjcyNDEzMTI6MTcyNjIzMjIwMQ==

**Comment by DurgeDriven** [comment]
I just use a old version of Explorer Patcher 
Fresh OS ( W11 Pro )  *** Do offline install using OOBE\BYPASSNRO
Block the 2 folders as described by others  ( I use placebos in my "Copy to C\: archive" )
Install patcher
Go straight to it and disable auto updates.
All she wrote  :) 

If the features work you want simply don't update it. 

![Screenshot 2024-09-15 081200](https://github.com/user-attachments/assets/3c5dc0bd-a259-409d-9e69-30efdcee210c)

![Screenshot 2024-09-15 083154](https://github.com/user-attachments/assets/88082fbd-7f99-44cc-a0a8-629c6727e6e3)

**Comment by DurgeDriven** [comment]
Just as a follow up someone from, here sent me a spam email with spool about totalvirus virustotal whatever sorry I  never responded to it. 

They said ..yadda  old version is no good look total virus finds all these, etc.

1.  My way I have no threats even detected, is totally clean ?
2. Why do I want to see false positives ?  
3. I use Malwarebytes Premium which just ignores it through the exclusions I set in Defender when I first do a fresh OS.
4. If you then disable explorer patcher updates straight away there is simply no way it can be detected ? 
5.  When I first setup explorer patcher Defender does absolutely nothing

I double checked 5 times different towers and OSes install , when I close Malwarebytes and check Defender info there is nothing
no warnings or anything else. 

HOWEVER ..........
IF I use recent explorer patcher setup exe and do exact same thing I get all the warnings and threat messages then have to block it.
I don't want to have to do that and my way is the only way I found possible.
Try a old version.

Peace

**Reply by pyrates999** [reply]
Read release notes: https://github.com/valinet/ExplorerPatcher/releases
